src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Project Duke is an educational software project designed to take you through
+ * the steps of building a small software incrementally, while applying as many
+ * Java and SE techniques as possible along the way.
+ *
+ * @author Bo Kuan (LG17)
+ * @version CS2103T AY22/23 Semester 2
+ */
+
+public class Duke {
+    /**
+     * The <code>Storage</code> object Duke accesses.
+     */
+    private Storage storage;
+    /**
+     * The <code>TaskList</code> object Duke accesses.
+     */
+    private TaskList tasks;
+    /**
+     * The <code>Ui</code> object Duke accesses.
+     */
+    private Ui ui;
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param filePath Filepath of the storage txt file
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts Duke's backend processes.
+     */
+    public void run() {
+        ui.showInitMessage();
+        ui.showGreeting();
```

Comment:
Is there a reason why you separated the InitMessage and Greeting into separate functions?

Reply:
I didn’t have a particular reason, but after the addition of GUI, I may not want the init message to be said by duke anymore as it isn’t really ‘speakable’